*** easeljs.js	Sat Sep 23 17:22:32 2023
--- easeljs-patched.js	Sat Sep 23 17:22:31 2023
***************
*** 1,4 ****
! var createjs = (this.createjs = (this.createjs || {}));
  /*!
  * EaselJS
  * Visit http://createjs.com/ for documentation, updates and examples.
--- 1,6 ----
! // var createjs = (this.createjs = (this.createjs || {}));
! var createjs = {};
! var ww; try { ww = window} catch { ww = false }
  /*!
  * EaselJS
  * Visit http://createjs.com/ for documentation, updates and examples.
***************
*** 32,38 ****
  // extend.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
--- 34,40 ----
  // extend.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
***************
*** 68,74 ****
  // promote.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
--- 70,76 ----
  // promote.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
***************
*** 128,134 ****
  // indexOf.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
--- 130,136 ----
  // indexOf.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
***************
*** 147,169 ****
   */
  createjs.indexOf = function (array, searchElement){
  	"use strict";
! 
! 	for (var i = 0,l=array.length; i < l; i++) {
! 		if (searchElement === array[i]) {
! 			return i;
! 		}
! 	}
! 	return -1;
  };
  
  //##############################################################################
  // UID.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 149,171 ----
   */
  createjs.indexOf = function (array, searchElement){
  	"use strict";
! 	return array.indexOf(searchElement);
! 	// for (var i = 0,l=array.length; i < l; i++) {
! 	// 	if (searchElement === array[i]) {
! 	// 		return i;
! 	// 	}
! 	// }
! 	// return -1;
  };
  
  //##############################################################################
  // UID.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! // 	"use strict";
  
  
  // constructor:
***************
*** 173,179 ****
  	 * @class UID
  	 * @static
  	 **/
! 	function UID() {
  		throw "UID cannot be instantiated";
  	}
  
--- 175,181 ----
  	 * @class UID
  	 * @static
  	 **/
! export function UID() {
  		throw "UID cannot be instantiated";
  	}
  
***************
*** 200,212 ****
  
  
  	createjs.UID = UID;
! }());
  
  //##############################################################################
  // deprecate.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
--- 202,214 ----
  
  
  	createjs.UID = UID;
! // }());
  
  //##############################################################################
  // deprecate.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
  /**
   * @class Utility Methods
***************
*** 248,257 ****
  // Event.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  // constructor:
  	/**
--- 250,259 ----
  // Event.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  // constructor:
  	/**
***************
*** 266,272 ****
  	 * @param {Boolean} cancelable Indicates whether the default behaviour of this event can be cancelled.
  	 * @constructor
  	 **/
! 	function Event(type, bubbles, cancelable) {
  		
  	
  	// public properties:
--- 268,274 ----
  	 * @param {Boolean} cancelable Indicates whether the default behaviour of this event can be cancelled.
  	 * @constructor
  	 **/
! export function Event(type, bubbles, cancelable) {
  		
  	
  	// public properties:
***************
*** 456,471 ****
  	};
  
  	createjs.Event = Event;
! }());
  
  //##############################################################################
  // EventDispatcher.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 458,473 ----
  	};
  
  	createjs.Event = Event;
! // }());
  
  //##############################################################################
  // EventDispatcher.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 523,529 ****
  	 * @class EventDispatcher
  	 * @constructor
  	 **/
! 	function EventDispatcher() {
  	
  	
  	// private properties:
--- 525,531 ----
  	 * @class EventDispatcher
  	 * @constructor
  	 **/
! export function EventDispatcher() {
  	
  	
  	// private properties:
***************
*** 556,561 ****
--- 558,564 ----
  	 * prototype.
  	 **/
  	EventDispatcher.initialize = function(target) {
+ 		var p = EventDispatcher.prototype;   // after modulization: var p gets otherwise bound below...
  		target.addEventListener = p.addEventListener;
  		target.on = p.on;
  		target.removeEventListener = target.off =  p.removeEventListener;
***************
*** 630,636 ****
  	 * @param {Function | Object} listener An object with a handleEvent method, or a function that will be called when
  	 * the event is dispatched.
  	 * @param {Object} [scope] The scope to execute the listener in. Defaults to the dispatcher/currentTarget for function listeners, and to the listener itself for object listeners (ie. using handleEvent).
! 	 * @param {Boolean} [once=false] If true, the listener will remove itself after the first time it is triggered.
  	 * @param {*} [data] Arbitrary data that will be included as the second parameter when the listener is called.
  	 * @param {Boolean} [useCapture=false] For events that bubble, indicates whether to listen for the event in the capture or bubbling/target phase.
  	 * @return {Function} Returns the anonymous function that was created and assigned as the listener. This is needed to remove the listener later using .removeEventListener.
--- 633,639 ----
  	 * @param {Function | Object} listener An object with a handleEvent method, or a function that will be called when
  	 * the event is dispatched.
  	 * @param {Object} [scope] The scope to execute the listener in. Defaults to the dispatcher/currentTarget for function listeners, and to the listener itself for object listeners (ie. using handleEvent).
! 	 * @param {Boolean} [once=false] If true, the listener is removed before the first time it is triggered.
  	 * @param {*} [data] Arbitrary data that will be included as the second parameter when the listener is called.
  	 * @param {Boolean} [useCapture=false] For events that bubble, indicates whether to listen for the event in the capture or bubbling/target phase.
  	 * @return {Function} Returns the anonymous function that was created and assigned as the listener. This is needed to remove the listener later using .removeEventListener.
***************
*** 641,650 ****
  			listener = listener.handleEvent;
  		}
  		scope = scope||this;
! 		return this.addEventListener(type, function(evt) {
! 				listener.call(scope, evt, data);
! 				once&&evt.remove();
! 			}, useCapture);
  	};
  
  	/**
--- 644,654 ----
  			listener = listener.handleEvent;
  		}
  		scope = scope||this;
! 		let d=this,listnr = function (evt) {
! 			once&&d.removeEventListener(type, listnr, useCapture)
! 			listener.call(scope, evt, data);
! 		}
! 		return this.addEventListener(type, listnr, useCapture);
  	};
  
  	/**
***************
*** 837,852 ****
  
  
  	createjs.EventDispatcher = EventDispatcher;
! }());
  
  //##############################################################################
  // Ticker.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 841,856 ----
  
  
  	createjs.EventDispatcher = EventDispatcher;
! // }());
  
  //##############################################################################
  // Ticker.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 872,878 ****
  	 * @uses EventDispatcher
  	 * @static
  	 **/
! 	function Ticker() {
  		throw "Ticker cannot be instantiated.";
  	}
  
--- 876,882 ----
  	 * @uses EventDispatcher
  	 * @static
  	 **/
! export function Ticker() {
  		throw "Ticker cannot be instantiated.";
  	}
  
***************
*** 1413,1435 ****
  	 * @static
  	 * @private
  	 **/
! 	var w=window, now=w.performance.now || w.performance.mozNow || w.performance.msNow || w.performance.oNow || w.performance.webkitNow;
  	Ticker._getTime = function() {
  		return ((now&&now.call(w.performance))||(new Date().getTime())) - Ticker._startTime;
  	};
  
  
  	createjs.Ticker = Ticker;
! }());
  
  //##############################################################################
  // VideoBuffer.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 1417,1439 ----
  	 * @static
  	 * @private
  	 **/
! 	 var w=ww&&window, now=ww&&(w.performance.now || w.performance.mozNow || w.performance.msNow || w.performance.oNow || w.performance.webkitNow);
  	Ticker._getTime = function() {
  		return ((now&&now.call(w.performance))||(new Date().getTime())) - Ticker._startTime;
  	};
  
  
  	createjs.Ticker = Ticker;
! // }());
  
  //##############################################################################
  // VideoBuffer.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 1445,1451 ****
  	 * @param {HTMLVideoElement} video The HTML video element to buffer.
  	 * @constructor
  	 **/
! 	function VideoBuffer(video) {
  		
  	// private properties:
  		/**
--- 1449,1455 ----
  	 * @param {HTMLVideoElement} video The HTML video element to buffer.
  	 * @constructor
  	 **/
! export function VideoBuffer(video) {
  		
  	// private properties:
  		/**
***************
*** 1520,1535 ****
  
  
  	createjs.VideoBuffer = VideoBuffer;
! }());
  
  //##############################################################################
  // MouseEvent.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 1524,1539 ----
  
  
  	createjs.VideoBuffer = VideoBuffer;
! // }());
  
  //##############################################################################
  // MouseEvent.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 1551,1557 ****
  	 * @extends Event
  	 * @constructor
  	 **/
! 	function MouseEvent(type, bubbles, cancelable, stageX, stageY, nativeEvent, pointerID, primary, rawX, rawY, relatedTarget) {
  		this.Event_constructor(type, bubbles, cancelable);
  		
  		
--- 1555,1561 ----
  	 * @extends Event
  	 * @constructor
  	 **/
! export function MouseEvent(type, bubbles, cancelable, stageX, stageY, nativeEvent, pointerID, primary, rawX, rawY, relatedTarget) {
  		this.Event_constructor(type, bubbles, cancelable);
  		
  		
***************
*** 1691,1706 ****
  
  
  	createjs.MouseEvent = createjs.promote(MouseEvent, "Event");
! }());
  
  //##############################################################################
  // Matrix2D.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 1695,1710 ----
  
  
  	createjs.MouseEvent = createjs.promote(MouseEvent, "Event");
! // }());
  
  //##############################################################################
  // Matrix2D.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 1724,1730 ****
  	 * @param {Number} [ty=0] Specifies the ty property for the new matrix.
  	 * @constructor
  	 **/
! 	function Matrix2D(a, b, c, d, tx, ty) {
  		this.setValues(a,b,c,d,tx,ty);
  		
  	// public properties:
--- 1728,1734 ----
  	 * @param {Number} [ty=0] Specifies the ty property for the new matrix.
  	 * @constructor
  	 **/
! export function Matrix2D(a, b, c, d, tx, ty) {
  		this.setValues(a,b,c,d,tx,ty);
  		
  	// public properties:
***************
*** 2193,2208 ****
  
  
  	createjs.Matrix2D = Matrix2D;
! }());
  
  //##############################################################################
  // DisplayProps.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  	/**
  	 * Used for calculating and encapsulating display related properties.
--- 2197,2212 ----
  
  
  	createjs.Matrix2D = Matrix2D;
! // }());
  
  //##############################################################################
  // DisplayProps.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! //(function() {
! //	"use strict";
  
  	/**
  	 * Used for calculating and encapsulating display related properties.
***************
*** 2214,2220 ****
  	 * @param {Number} [matrix] A transformation matrix. Defaults to a new identity matrix.
  	 * @constructor
  	 **/
! 	function DisplayProps(visible, alpha, shadow, compositeOperation, matrix) {
  		this.setValues(visible, alpha, shadow, compositeOperation, matrix);
  		
  	// public properties:
--- 2218,2224 ----
  	 * @param {Number} [matrix] A transformation matrix. Defaults to a new identity matrix.
  	 * @constructor
  	 **/
! export function DisplayProps(visible, alpha, shadow, compositeOperation, matrix) {
  		this.setValues(visible, alpha, shadow, compositeOperation, matrix);
  		
  	// public properties:
***************
*** 2350,2365 ****
  // private methods:
  
  	createjs.DisplayProps = DisplayProps;
! })();
  
  //##############################################################################
  // Point.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 2354,2369 ----
  // private methods:
  
  	createjs.DisplayProps = DisplayProps;
! //})();
  
  //##############################################################################
  // Point.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 2375,2381 ****
  	 * @param {Number} [y=0] Y position.
  	 * @constructor
  	 **/
! 	function Point(x, y) {
  	 	this.setValues(x, y);
  	 	
  	 	
--- 2379,2385 ----
  	 * @param {Number} [y=0] Y position.
  	 * @constructor
  	 **/
! export function Point(x, y) {
  	 	this.setValues(x, y);
  	 	
  	 	
***************
*** 2443,2458 ****
  	
  	
  	createjs.Point = Point;
! }());
  
  //##############################################################################
  // Rectangle.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 2447,2462 ----
  	
  	
  	createjs.Point = Point;
! // }());
  
  //##############################################################################
  // Rectangle.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 2470,2476 ****
  	 * @param {Number} [height=0] The height of the Rectangle.
  	 * @constructor
  	 **/
! 	function Rectangle(x, y, width, height) {
  		this.setValues(x, y, width, height);
  		
  		
--- 2474,2480 ----
  	 * @param {Number} [height=0] The height of the Rectangle.
  	 * @constructor
  	 **/
! export function Rectangle(x, y, width, height) {
  		this.setValues(x, y, width, height);
  		
  		
***************
*** 2651,2666 ****
  	
  	
  	createjs.Rectangle = Rectangle;
! }());
  
  //##############################################################################
  // ButtonHelper.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 2655,2670 ----
  	
  	
  	createjs.Rectangle = Rectangle;
! // }());
  
  //##############################################################################
  // ButtonHelper.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 2697,2703 ****
  	 * null, then the default state of the hitArea will be used. *
  	 * @constructor
  	 */
! 	function ButtonHelper(target, outLabel, overLabel, downLabel, play, hitArea, hitLabel) {
  		if (!target.addEventListener) { return; }
  	
  	
--- 2701,2707 ----
  	 * null, then the default state of the hitArea will be used. *
  	 * @constructor
  	 */
! export function ButtonHelper(target, outLabel, overLabel, downLabel, play, hitArea, hitLabel) {
  		if (!target.addEventListener) { return; }
  	
  	
***************
*** 2884,2899 ****
  
  
  	createjs.ButtonHelper = ButtonHelper;
! }());
  
  //##############################################################################
  // Shadow.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 2888,2903 ----
  
  
  	createjs.ButtonHelper = ButtonHelper;
! // }());
  
  //##############################################################################
  // Shadow.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 2912,2918 ****
  	 * @param {Number} offsetY The y offset of the shadow in pixels.
  	 * @param {Number} blur The size of the blurring effect.
  	 **/
! 	function Shadow(color, offsetX, offsetY, blur) {
  		
  		
  	// public properties:
--- 2916,2922 ----
  	 * @param {Number} offsetY The y offset of the shadow in pixels.
  	 * @param {Number} blur The size of the blurring effect.
  	 **/
! export function Shadow(color, offsetX, offsetY, blur) {
  		
  		
  	// public properties:
***************
*** 2980,2995 ****
  	
  
  	createjs.Shadow = Shadow;
! }());
  
  //##############################################################################
  // SpriteSheet.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 2984,2999 ----
  	
  
  	createjs.Shadow = Shadow;
! // }());
  
  //##############################################################################
  // SpriteSheet.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 3161,3167 ****
  	 * @param {Object} data An object describing the SpriteSheet data.
  	 * @extends EventDispatcher
  	 **/
! 	function SpriteSheet(data) {
  		this.EventDispatcher_constructor();
  
  
--- 3165,3171 ----
  	 * @param {Object} data An object describing the SpriteSheet data.
  	 * @extends EventDispatcher
  	 **/
! export function SpriteSheet(data) {
  		this.EventDispatcher_constructor();
  
  
***************
*** 3579,3594 ****
  
  
  	createjs.SpriteSheet = createjs.promote(SpriteSheet, "EventDispatcher");
! }());
  
  //##############################################################################
  // Graphics.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 3583,3598 ----
  
  
  	createjs.SpriteSheet = createjs.promote(SpriteSheet, "EventDispatcher");
! // }());
  
  //##############################################################################
  // Graphics.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 3687,3693 ****
  	 * @class Graphics
  	 * @constructor
  	 **/
! 	function Graphics() {
  
  
  	// public properties
--- 3691,3697 ----
  	 * @class Graphics
  	 * @constructor
  	 **/
! export function Graphics() {
  
  
  	// public properties
***************
*** 3922,3928 ****
  	 * @protected
  	 * @type {CanvasRenderingContext2D}
  	 **/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():document.createElement("canvas"));
  	if (canvas.getContext) {
  		Graphics._ctx = canvas.getContext("2d");
  		canvas.width = canvas.height = 1;
--- 3926,3932 ----
  	 * @protected
  	 * @type {CanvasRenderingContext2D}
  	 **/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():ww&&document.createElement("canvas"));
  	if (canvas.getContext) {
  		Graphics._ctx = canvas.getContext("2d");
  		canvas.width = canvas.height = 1;
***************
*** 5931,5946 ****
  
  
  	createjs.Graphics = Graphics;
! }());
  
  //##############################################################################
  // DisplayObject.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 5935,5950 ----
  
  
  	createjs.Graphics = Graphics;
! // }());
  
  //##############################################################################
  // DisplayObject.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 5954,5960 ****
  	 * @extends EventDispatcher
  	 * @constructor
  	 **/
! 	function DisplayObject() {
  		this.EventDispatcher_constructor();
  
  
--- 5958,5964 ----
  	 * @extends EventDispatcher
  	 * @constructor
  	 **/
! export function DisplayObject() {
  		this.EventDispatcher_constructor();
  
  
***************
*** 6346,6352 ****
  	 * @static
  	 * @protected
  	 **/
! 	var canvas = createjs.createCanvas?createjs.createCanvas():document.createElement("canvas"); // prevent errors on load in browsers without canvas.
  	if (canvas.getContext) {
  		DisplayObject._hitTestCanvas = canvas;
  		DisplayObject._hitTestContext = canvas.getContext("2d");
--- 6350,6356 ----
  	 * @static
  	 * @protected
  	 **/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():ww&&document.createElement("canvas")); // prevent errors on load in browsers without canvas.
  	if (canvas.getContext) {
  		DisplayObject._hitTestCanvas = canvas;
  		DisplayObject._hitTestContext = canvas.getContext("2d");
***************
*** 7182,7197 ****
  	};
  
  	createjs.DisplayObject = createjs.promote(DisplayObject, "EventDispatcher");
! }());
  
  //##############################################################################
  // Container.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  	
  
  // constructor:
--- 7186,7201 ----
  	};
  
  	createjs.DisplayObject = createjs.promote(DisplayObject, "EventDispatcher");
! // }());
  
  //##############################################################################
  // Container.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  	
  
  // constructor:
***************
*** 7216,7222 ****
   * @extends DisplayObject
   * @constructor
   **/
! 	function Container() {
  		this.DisplayObject_constructor();
  		
  	// public properties:
--- 7220,7226 ----
   * @extends DisplayObject
   * @constructor
   **/
! export function Container() {
  		this.DisplayObject_constructor();
  		
  	// public properties:
***************
*** 7374,7385 ****
  	 *
  	 *      addChildAt(child1, child2, ..., index);
  	 *
! 	 * The index must be between 0 and numChildren. For example, to add myShape under otherShape in the display list,
! 	 * you could use:
  	 *
  	 *      container.addChildAt(myShape, container.getChildIndex(otherShape));
  	 *
! 	 * This would also bump otherShape's index up by one. Fails silently if the index is out of range.
  	 *
  	 * @method addChildAt
  	 * @param {DisplayObject} child The display object to add.
--- 7378,7390 ----
  	 *
  	 *      addChildAt(child1, child2, ..., index);
  	 *
! 	 * For example, to add myShape under otherShape in the display list, you could use:
  	 *
  	 *      container.addChildAt(myShape, container.getChildIndex(otherShape));
  	 *
! 	 * This would also bump otherShape's index up by one.
! 	 * 
! 	 * The index works like the first parameter of splice(), so -1 inserts before the last child.
  	 *
  	 * @method addChildAt
  	 * @param {DisplayObject} child The display object to add.
***************
*** 7866,7881 ****
  
  
  	createjs.Container = createjs.promote(Container, "DisplayObject");
! }());
  
  //##############################################################################
  // Stage.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 7871,7886 ----
  
  
  	createjs.Container = createjs.promote(Container, "DisplayObject");
! // }());
  
  //##############################################################################
  // Stage.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 7902,7908 ****
  	 * @param {HTMLCanvasElement | String | Object} canvas A canvas object that the Stage will render to, or the string id
  	 * of a canvas object in the current document.
  	 **/
! 	function Stage(canvas) {
  		this.Container_constructor();
  	
  	
--- 7907,7913 ----
  	 * @param {HTMLCanvasElement | String | Object} canvas A canvas object that the Stage will render to, or the string id
  	 * of a canvas object in the current document.
  	 **/
! export function Stage(canvas) {
  		this.Container_constructor();
  	
  	
***************
*** 7937,7943 ****
  		 * @property canvas
  		 * @type HTMLCanvasElement | Object
  		 **/
! 		this.canvas = (typeof canvas == "string") ? document.getElementById(canvas) : canvas;
  	
  		/**
  		 * The current mouse X position on the canvas. If the mouse leaves the canvas, this will indicate the most recent
--- 7942,7948 ----
  		 * @property canvas
  		 * @type HTMLCanvasElement | Object
  		 **/
! 		this.canvas = (typeof canvas == "string") ? ww&&document.getElementById(canvas) : canvas;
  	
  		/**
  		 * The current mouse X position on the canvas. If the mouse leaves the canvas, this will indicate the most recent
***************
*** 8608,8613 ****
--- 8613,8619 ----
  			nextStage&&nextStage._testMouseOver(clear, owner, eventTarget);
  			return;
  		}
+ 		if (this.stage.canvas) {
  		var o = this._getPointerData(-1);
  		// only update if the mouse position has changed. This provides a lot of optimization, but has some trade-offs.
  		if (!o || (!clear && this.mouseX == this._mouseOverX && this.mouseY == this._mouseOverY && this.mouseInBounds)) { return; }
***************
*** 8657,8662 ****
--- 8663,8669 ----
  		if (oldTarget != target) {
  			this._dispatchMouseEvent(target, "mouseover", true, -1, o, e, oldTarget);
  		}
+ 		}
  		
  		nextStage&&nextStage._testMouseOver(clear, owner || target && this, eventTarget || isEventTarget && this);
  	};
***************
*** 8702,8714 ****
  
  
  	createjs.Stage = createjs.promote(Stage, "Container");
! }());
  
  //##############################################################################
  // StageGL.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
  /*
   * README IF EDITING:
--- 8709,8721 ----
  
  
  	createjs.Stage = createjs.promote(Stage, "Container");
! // }());
  
  //##############################################################################
  // StageGL.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
  /*
   * README IF EDITING:
***************
*** 8728,8735 ****
   * Slot: A space on the GPU into which textures can be loaded for use in a batch, using "ActiveTexture" switches texture slot.
   */
  
! (function () {
! 	"use strict";
  
  	/**
  	 * A StageGL instance is the root level {{#crossLink "Container"}}{{/crossLink}} for an WebGL-optimized display list,
--- 8735,8742 ----
   * Slot: A space on the GPU into which textures can be loaded for use in a batch, using "ActiveTexture" switches texture slot.
   */
  
! // (function () {
! //	"use strict";
  
  	/**
  	 * A StageGL instance is the root level {{#crossLink "Container"}}{{/crossLink}} for an WebGL-optimized display list,
***************
*** 8797,8803 ****
  	 * `purgeTextures(autoPurge)` every `autoPurge/2` draws. See {{#crossLink "StageGL/purgeTextures"}}{{/crossLink}} for more
  	 * information.
  	 */
! 	function StageGL(canvas, options) {
  		this.Stage_constructor(canvas);
  
  		if (options !== undefined) {
--- 8804,8810 ----
  	 * `purgeTextures(autoPurge)` every `autoPurge/2` draws. See {{#crossLink "StageGL/purgeTextures"}}{{/crossLink}} for more
  	 * information.
  	 */
! export function StageGL(canvas, options) {
  		this.Stage_constructor(canvas);
  
  		if (options !== undefined) {
***************
*** 11101,11115 ****
  	};
  
  	createjs.StageGL = createjs.promote(StageGL, "Stage");
! }());
  
  //##############################################################################
  // Bitmap.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
  	
  	/**
  	 * A Bitmap represents an Image, Canvas, or Video in the display list. A Bitmap can be instantiated using an existing
--- 11108,11122 ----
  	};
  
  	createjs.StageGL = createjs.promote(StageGL, "Stage");
! // }());
  
  //##############################################################################
  // Bitmap.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
  	
  	/**
  	 * A Bitmap represents an Image, Canvas, or Video in the display list. A Bitmap can be instantiated using an existing
***************
*** 11141,11147 ****
  	 * (image, video, canvas), an object with a `getImage` method that returns a CanvasImageSource, or a string URL to an image.
  	 * If the latter, a new Image instance with the URL as its src will be used.
  	 **/
! 	function Bitmap(imageOrUri) {
  		this.DisplayObject_constructor();
  		
  		
--- 11148,11154 ----
  	 * (image, video, canvas), an object with a `getImage` method that returns a CanvasImageSource, or a string URL to an image.
  	 * If the latter, a new Image instance with the URL as its src will be used.
  	 **/
! export function Bitmap(imageOrUri) {
  		this.DisplayObject_constructor();
  		
  		
***************
*** 11306,11321 ****
  
  	
  	createjs.Bitmap = createjs.promote(Bitmap, "DisplayObject");
! }());
  
  //##############################################################################
  // Sprite.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 11313,11328 ----
  
  	
  	createjs.Bitmap = createjs.promote(Bitmap, "DisplayObject");
! // }());
  
  //##############################################################################
  // Sprite.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 11342,11348 ****
  	 * dimensions, and frame data. See {{#crossLink "SpriteSheet"}}{{/crossLink}} for more information.
  	 * @param {String|Number} [frameOrAnimation] The frame number or animation to play initially.
  	 **/
! 	function Sprite(spriteSheet, frameOrAnimation) {
  		this.DisplayObject_constructor();
  		
  		
--- 11349,11355 ----
  	 * dimensions, and frame data. See {{#crossLink "SpriteSheet"}}{{/crossLink}} for more information.
  	 * @param {String|Number} [frameOrAnimation] The frame number or animation to play initially.
  	 **/
! export function Sprite(spriteSheet, frameOrAnimation) {
  		this.DisplayObject_constructor();
  		
  		
***************
*** 11759,11774 ****
  
  
  	createjs.Sprite = createjs.promote(Sprite, "DisplayObject");
! }());
  
  //##############################################################################
  // Shape.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 11766,11781 ----
  
  
  	createjs.Sprite = createjs.promote(Sprite, "DisplayObject");
! // }());
  
  //##############################################################################
  // Shape.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 11795,11801 ****
  	 * @constructor
  	 * @param {Graphics} graphics Optional. The graphics instance to display. If null, a new Graphics instance will be created.
  	 **/
! 	function Shape(graphics) {
  		this.DisplayObject_constructor();
  		
  		
--- 11802,11808 ----
  	 * @constructor
  	 * @param {Graphics} graphics Optional. The graphics instance to display. If null, a new Graphics instance will be created.
  	 **/
! export function Shape(graphics) {
  		this.DisplayObject_constructor();
  		
  		
***************
*** 11866,11881 ****
  
  
  	createjs.Shape = createjs.promote(Shape, "DisplayObject");
! }());
  
  //##############################################################################
  // Text.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 11873,11888 ----
  
  
  	createjs.Shape = createjs.promote(Shape, "DisplayObject");
! // }());
  
  //##############################################################################
  // Text.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 11908,11914 ****
  	 * @param {String} [color] The color to draw the text in. Any valid value for the CSS color attribute is acceptable (ex.
  	 * "#F00", "red", or "#FF0000").
  	 **/
! 	function Text(text, font, color) {
  		this.DisplayObject_constructor();
  		
  		
--- 11915,11921 ----
  	 * @param {String} [color] The color to draw the text in. Any valid value for the CSS color attribute is acceptable (ex.
  	 * "#F00", "red", or "#FF0000").
  	 **/
! export function Text(text, font, color) {
  		this.DisplayObject_constructor();
  		
  		
***************
*** 11999,12005 ****
  	 * @type CanvasRenderingContext2D
  	 * @private
  	 **/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():document.createElement("canvas"));
  	if (canvas.getContext) { Text._workingContext = canvas.getContext("2d"); canvas.width = canvas.height = 1; }
  	
  	
--- 12006,12012 ----
  	 * @type CanvasRenderingContext2D
  	 * @private
  	 **/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():ww&&document.createElement("canvas"));
  	if (canvas.getContext) { Text._workingContext = canvas.getContext("2d"); canvas.width = canvas.height = 1; }
  	
  	
***************
*** 12261,12276 ****
  
  
  	createjs.Text = createjs.promote(Text, "DisplayObject");
! }());
  
  //##############################################################################
  // BitmapText.js
  //##############################################################################
  
! this.createjs = this.createjs || {};
  
! (function () {
! 	"use strict";
  
  
  // constructor:
--- 12268,12283 ----
  
  
  	createjs.Text = createjs.promote(Text, "DisplayObject");
! // }());
  
  //##############################################################################
  // BitmapText.js
  //##############################################################################
  
! // this.createjs = this.createjs || {};
  
! // (function () {
! //	"use strict";
  
  
  // constructor:
***************
*** 12289,12295 ****
  	 * @param {SpriteSheet} [spriteSheet=null] The spritesheet that defines the character glyphs.
  	 * @constructor
  	 **/
! 	function BitmapText(text, spriteSheet) {
  		this.Container_constructor();
  		
  		
--- 12296,12302 ----
  	 * @param {SpriteSheet} [spriteSheet=null] The spritesheet that defines the character glyphs.
  	 * @constructor
  	 **/
! export function BitmapText(text, spriteSheet) {
  		this.Container_constructor();
  		
  		
***************
*** 12592,12607 ****
  
  
  	createjs.BitmapText = createjs.promote(BitmapText, "Container");
! }());
  
  //##############################################################################
  // MovieClip.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 12599,12614 ----
  
  
  	createjs.BitmapText = createjs.promote(BitmapText, "Container");
! // }());
  
  //##############################################################################
  // MovieClip.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 12658,12664 ****
  	 * @extends Container
  	 * @constructor
  	 **/
! 	function MovieClip(props) {
  		this.Container_constructor();
  		!MovieClip.inited&&MovieClip.init(); // static init
  		
--- 12665,12671 ----
  	 * @extends Container
  	 * @constructor
  	 **/
! export function MovieClip(props) {
  		this.Container_constructor();
  		!MovieClip.inited&&MovieClip.init(); // static init
  		
***************
*** 13322,13337 ****
  		if (prop === "startPosition") { return (ratio === 1 ? step.props[prop] : step.prev.props[prop]); }
  	};
  
! }());
  
  //##############################################################################
  // SpriteSheetUtils.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  	
  	
  // constructor:
--- 13329,13344 ----
  		if (prop === "startPosition") { return (ratio === 1 ? step.props[prop] : step.prev.props[prop]); }
  	};
  
! // }());
  
  //##############################################################################
  // SpriteSheetUtils.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  	
  	
  // constructor:
***************
*** 13343,13349 ****
  	 * @class SpriteSheetUtils
  	 * @static
  	 **/
! 	function SpriteSheetUtils() {
  		throw "SpriteSheetUtils cannot be instantiated";
  	}
  
--- 13350,13356 ----
  	 * @class SpriteSheetUtils
  	 * @static
  	 **/
! export function SpriteSheetUtils() {
  		throw "SpriteSheetUtils cannot be instantiated";
  	}
  
***************
*** 13361,13367 ****
  	 * @type CanvasRenderingContext2D
  	 * @protected
  	*/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():document.createElement("canvas"));
  	if (canvas.getContext) {
  		SpriteSheetUtils._workingCanvas = canvas;
  		SpriteSheetUtils._workingContext = canvas.getContext("2d");
--- 13368,13374 ----
  	 * @type CanvasRenderingContext2D
  	 * @protected
  	*/
! 	var canvas = (createjs.createCanvas?createjs.createCanvas():ww&&document.createElement("canvas"));
  	if (canvas.getContext) {
  		SpriteSheetUtils._workingCanvas = canvas;
  		SpriteSheetUtils._workingContext = canvas.getContext("2d");
***************
*** 13472,13487 ****
  
  
  	createjs.SpriteSheetUtils = SpriteSheetUtils;
! }());
  
  //##############################################################################
  // SpriteSheetBuilder.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 13479,13494 ----
  
  
  	createjs.SpriteSheetUtils = SpriteSheetUtils;
! // }());
  
  //##############################################################################
  // SpriteSheetBuilder.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 13502,13508 ****
  	 * @extends EventDispatcher
  	 * @constructor
  	 **/
! 	function SpriteSheetBuilder(framerate) {
  		this.EventDispatcher_constructor();
  		
  	// public properties:
--- 13509,13515 ----
  	 * @extends EventDispatcher
  	 * @constructor
  	 **/
! export function SpriteSheetBuilder(framerate) {
  		this.EventDispatcher_constructor();
  		
  	// public properties:
***************
*** 13848,13854 ****
  			if (o.w > x) { x = o.w; }
  			y += o.h;
  			if (!o.h || !frames.length) {
! 				var canvas = createjs.createCanvas?createjs.createCanvas():document.createElement("canvas");
  				canvas.width = this._getSize(x,this.maxWidth);
  				canvas.height = this._getSize(y,this.maxHeight);
  				this._data.images[img] = canvas;
--- 13855,13861 ----
  			if (o.w > x) { x = o.w; }
  			y += o.h;
  			if (!o.h || !frames.length) {
! 				var canvas = (createjs.createCanvas?createjs.createCanvas():ww&&document.createElement("canvas"));
  				canvas.width = this._getSize(x,this.maxWidth);
  				canvas.height = this._getSize(y,this.maxHeight);
  				this._data.images[img] = canvas;
***************
*** 13984,13999 ****
  
  
  	createjs.SpriteSheetBuilder = createjs.promote(SpriteSheetBuilder, "EventDispatcher");
! }());
  
  //##############################################################################
  // DOMElement.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 13991,14006 ----
  
  
  	createjs.SpriteSheetBuilder = createjs.promote(SpriteSheetBuilder, "EventDispatcher");
! // }());
  
  //##############################################################################
  // DOMElement.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 14033,14039 ****
  	 * @constructor
  	 * @param {HTMLElement} htmlElement A reference or id for the DOM element to manage.
  	 */
! 	function DOMElement(htmlElement) {
  		this.DisplayObject_constructor();
  		
  		if (typeof(htmlElement)=="string") { htmlElement = document.getElementById(htmlElement); }
--- 14040,14046 ----
  	 * @constructor
  	 * @param {HTMLElement} htmlElement A reference or id for the DOM element to manage.
  	 */
! export function DOMElement(htmlElement) {
  		this.DisplayObject_constructor();
  		
  		if (typeof(htmlElement)=="string") { htmlElement = document.getElementById(htmlElement); }
***************
*** 14252,14267 ****
  
  
  	createjs.DOMElement = createjs.promote(DOMElement, "DisplayObject");
! }());
  
  //##############################################################################
  // Filter.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 14259,14274 ----
  
  
  	createjs.DOMElement = createjs.promote(DOMElement, "DisplayObject");
! // }());
  
  //##############################################################################
  // Filter.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 14294,14300 ****
  	 * @class Filter
  	 * @constructor
  	 **/
! 	function Filter() {
  		/**
  		 * A flag stating that this filter uses a context draw mode and cannot be batched into imageData processing.
  		 * @property usesContext
--- 14301,14307 ----
  	 * @class Filter
  	 * @constructor
  	 **/
! export function Filter() {
  		/**
  		 * A flag stating that this filter uses a context draw mode and cannot be batched into imageData processing.
  		 * @property usesContext
***************
*** 14413,14428 ****
  
  
  	createjs.Filter = Filter;
! }());
  
  //##############################################################################
  // BitmapCache.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 14420,14435 ----
  
  
  	createjs.Filter = Filter;
! // }());
  
  //##############################################################################
  // BitmapCache.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 14447,14453 ****
  	 * @class BitmapCache
  	 * @constructor
  	 **/
! 	function BitmapCache() {
  
  		// public:
  		/**
--- 14454,14460 ----
  	 * @class BitmapCache
  	 * @constructor
  	 **/
! export function BitmapCache() {
  
  		// public:
  		/**
***************
*** 14940,14955 ****
  	};
  
  	createjs.BitmapCache = BitmapCache;
! }());
  
  //##############################################################################
  // BlurFilter.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 14947,14962 ----
  	};
  
  	createjs.BitmapCache = BitmapCache;
! // }());
  
  //##############################################################################
  // BlurFilter.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 14978,14984 ****
  	 * @param {Number} [blurY=0] The vertical blur radius in pixels.
  	 * @param {Number} [quality=1] The number of blur iterations.
  	 **/
! 	function BlurFilter( blurX, blurY, quality) {
  		this.Filter_constructor();
  
  		// public properties:
--- 14985,14991 ----
  	 * @param {Number} [blurY=0] The vertical blur radius in pixels.
  	 * @param {Number} [quality=1] The number of blur iterations.
  	 **/
! export function BlurFilter(blurX, blurY, quality) {
  		this.Filter_constructor();
  
  		// public properties:
***************
*** 15387,15402 ****
  	};
  
  	createjs.BlurFilter = createjs.promote(BlurFilter, "Filter");
! }());
  
  //##############################################################################
  // AlphaMapFilter.js
  //##############################################################################
  
! this.createjs = this.createjs || {};
  
! (function () {
! 	"use strict";
  	
  	
  // constructor:
--- 15394,15409 ----
  	};
  
  	createjs.BlurFilter = createjs.promote(BlurFilter, "Filter");
! // }());
  
  //##############################################################################
  // AlphaMapFilter.js
  //##############################################################################
  
! // this.createjs = this.createjs || {};
  
! // (function () {
! //	"use strict";
  	
  	
  // constructor:
***************
*** 15429,15435 ****
  	 * @param {HTMLImageElement|HTMLCanvasElement} alphaMap The greyscale image (or canvas) to use as the alpha value for the
  	 * result. This should be exactly the same dimensions as the target.
  	 **/
! 	function AlphaMapFilter(alphaMap) {
  		this.Filter_constructor();
  	
  	// public properties:
--- 15436,15442 ----
  	 * @param {HTMLImageElement|HTMLCanvasElement} alphaMap The greyscale image (or canvas) to use as the alpha value for the
  	 * result. This should be exactly the same dimensions as the target.
  	 **/
! export function AlphaMapFilter(alphaMap) {
  		this.Filter_constructor();
  	
  	// public properties:
***************
*** 15555,15570 ****
  
  
  	createjs.AlphaMapFilter = createjs.promote(AlphaMapFilter, "Filter");
! }());
  
  //##############################################################################
  // AlphaMaskFilter.js
  //##############################################################################
  
! this.createjs = this.createjs || {};
  
! (function () {
! 	"use strict";
  
  
  // constructor:
--- 15562,15577 ----
  
  
  	createjs.AlphaMapFilter = createjs.promote(AlphaMapFilter, "Filter");
! // }());
  
  //##############################################################################
  // AlphaMaskFilter.js
  //##############################################################################
  
! // this.createjs = this.createjs || {};
  
! // (function () {
! //	"use strict";
  
  
  // constructor:
***************
*** 15596,15602 ****
  	 * @constructor
  	 * @param {HTMLImageElement|HTMLCanvasElement} mask
  	 **/
! 	function AlphaMaskFilter(mask) {
  		this.Filter_constructor();
  	
  	// public properties:
--- 15603,15609 ----
  	 * @constructor
  	 * @param {HTMLImageElement|HTMLCanvasElement} mask
  	 **/
! export function AlphaMaskFilter(mask) {
  		this.Filter_constructor();
  	
  	// public properties:
***************
*** 15689,15704 ****
  
  
  	createjs.AlphaMaskFilter = createjs.promote(AlphaMaskFilter, "Filter");
! }());
  
  //##############################################################################
  // ColorFilter.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 15696,15711 ----
  
  
  	createjs.AlphaMaskFilter = createjs.promote(AlphaMaskFilter, "Filter");
! // }());
  
  //##############################################################################
  // ColorFilter.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 15734,15740 ****
  	 * @constructor
  	 * @extends Filter
  	 **/
! 	function ColorFilter(redMultiplier, greenMultiplier, blueMultiplier, alphaMultiplier, redOffset, greenOffset, blueOffset, alphaOffset) {
  		this.Filter_constructor();
  
  	// public properties:
--- 15741,15747 ----
  	 * @constructor
  	 * @extends Filter
  	 **/
! export function ColorFilter(redMultiplier, greenMultiplier, blueMultiplier, alphaMultiplier, redOffset, greenOffset, blueOffset, alphaOffset) {
  		this.Filter_constructor();
  
  	// public properties:
***************
*** 15855,15870 ****
  
  
  	createjs.ColorFilter = createjs.promote(ColorFilter, "Filter");
! }());
  
  //##############################################################################
  // ColorMatrix.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 15862,15877 ----
  
  
  	createjs.ColorFilter = createjs.promote(ColorFilter, "Filter");
! // }());
  
  //##############################################################################
  // ColorMatrix.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 15885,15891 ****
  	 * @param {Number} hue
  	 * @constructor
  	 **/
! 	function ColorMatrix(brightness, contrast, saturation, hue) {
  		this.setColor(brightness, contrast, saturation, hue);
  	}
  	var p = ColorMatrix.prototype;
--- 15892,15898 ----
  	 * @param {Number} hue
  	 * @constructor
  	 **/
! export function ColorMatrix(brightness, contrast, saturation, hue) {
  		this.setColor(brightness, contrast, saturation, hue);
  	}
  	var p = ColorMatrix.prototype;
***************
*** 16199,16214 ****
  
  
  	createjs.ColorMatrix = ColorMatrix;
! }());
  
  //##############################################################################
  // ColorMatrixFilter.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 16206,16221 ----
  
  
  	createjs.ColorMatrix = ColorMatrix;
! // }());
  
  //##############################################################################
  // ColorMatrixFilter.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 16237,16243 ****
  	 * @param {Array | ColorMatrix} matrix A 4x5 matrix describing the color operation to perform. See also the {{#crossLink "ColorMatrix"}}{{/crossLink}}
  	 * class.
  	 **/
! 	function ColorMatrixFilter(matrix) {
  		this.Filter_constructor();
  
  	// public properties:
--- 16244,16250 ----
  	 * @param {Array | ColorMatrix} matrix A 4x5 matrix describing the color operation to perform. See also the {{#crossLink "ColorMatrix"}}{{/crossLink}}
  	 * class.
  	 **/
! export function ColorMatrixFilter(matrix) {
  		this.Filter_constructor();
  
  	// public properties:
***************
*** 16328,16343 ****
  	};
  
  	createjs.ColorMatrixFilter = createjs.promote(ColorMatrixFilter, "Filter");
! }());
  
  //##############################################################################
  // Touch.js
  //##############################################################################
  
! this.createjs = this.createjs||{};
  
! (function() {
! 	"use strict";
  
  
  // constructor:
--- 16335,16350 ----
  	};
  
  	createjs.ColorMatrixFilter = createjs.promote(ColorMatrixFilter, "Filter");
! // }());
  
  //##############################################################################
  // Touch.js
  //##############################################################################
  
! // this.createjs = this.createjs||{};
  
! // (function() {
! //	"use strict";
  
  
  // constructor:
***************
*** 16360,16366 ****
   * @class Touch
   * @static
   **/
! 	function Touch() {
  		throw "Touch cannot be instantiated";
  	}
  
--- 16367,16373 ----
   * @class Touch
   * @static
   **/
! export function Touch() {
  		throw "Touch cannot be instantiated";
  	}
  
***************
*** 16614,16626 ****
  
  
  	createjs.Touch = Touch;
! }());
  
  //##############################################################################
  // version.js
  //##############################################################################
  
! this.createjs = this.createjs || {};
  
  (function() {
  	"use strict";
--- 16621,16633 ----
  
  
  	createjs.Touch = Touch;
! // }());
  
  //##############################################################################
  // version.js
  //##############################################################################
  
! // this.createjs = this.createjs || {};
  
  (function() {
  	"use strict";
